c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -3625,6 +3630,388 @@ bool response_started(JSObject *self) {
 
 namespace GlobalProperties {
+
+JS::Result<std::string> ConvertJSValueToByteString(JSContext *cx, JS::Handle<JS::Value> v) {
+  JS::RootedString s(cx);
+  if (v.isString()) {
+    s = v.toString();
+  } else {
+    s = JS::ToString(cx, v);
+    if (!s) {
+      JS_ReportErrorNumberUTF8(cx, GetErrorMessage, nullptr, JSMSG_INVALID_CHARACTER_ERROR);
+      return JS::Result<std::string>(JS::Error());
+    }
+  }
+
+  // Conversion from Javascript string to ByteString is only valid if all
+  // characters < 256. This is always the case for Latin1 strings.
+  size_t length;
+  if (!JS::StringHasLatin1Chars(s)) {
+    // Creating an exception can GC, so we first scan the string for bad chars
+    // and report the error outside the AutoCheckCannotGC scope.
+    bool foundBadChar = false;
+    size_t badCharIndex;
+    char16_t badChar;
+    {
+      JS::AutoCheckCannotGC nogc;
+      const char16_t *chars = JS_GetTwoByteStringCharsAndLength(cx, nogc, s, &length);
+      if (!chars) {
+        JS_ReportErrorNumberUTF8(cx, GetErrorMessage, nullptr, JSMSG_INVALID_CHARACTER_ERROR);
+        return JS::Result<std::string>(JS::Error());
+      }
+
+      for (size_t i = 0; i < length; i++) {
+        if (chars[i] > 255) {
+          badCharIndex = i;
+          badChar = chars[i];
+          foundBadChar = true;
+          break;
+        }
+      }
+    }
+
+    if (foundBadChar) {
+      JS_ReportErrorNumberUTF8(cx, GetErrorMessage, nullptr, JSMSG_INVALID_CHARACTER_ERROR);
+      return JS::Result<std::string>(JS::Error());
+    }
+  } else {
+    length = JS::GetStringLength(s);
+  }
+
+  UniqueChars result = JS_EncodeStringToLatin1(cx, s);
+  if (!result) {
+    return JS::Result<std::string>(JS::Error());
+  }
+  std::string byteString(result.get(), length);
+  return byteString;
+}
+
+// Maps an encoded character to a value in the Base64 alphabet, per
+// RFC 4648, Table 1. Invalid input characters map to UINT8_MAX.
+// https://datatracker.ietf.org/doc/html/rfc4648#section-4
+
+static const uint8_t base64DecodeTable[] = {
+    // clang-format off
+  /* 0 */  255, 255, 255, 255, 255, 255, 255, 255,
+  /* 8 */  255, 255, 255, 255, 255, 255, 255, 255,
+  /* 16 */ 255, 255, 255, 255, 255, 255, 255, 255,
+  /* 24 */ 255, 255, 255, 255, 255, 255, 255, 255,
+  /* 32 */ 255, 255, 255, 255, 255, 255, 255, 255,
+  /* 40 */ 255, 255, 255,
+  62 /* + */,
+  255, 255, 255,
+  63 /* / */,
+
+  /* 48 */ /* 0 - 9 */ 52, 53, 54, 55, 56, 57, 58, 59,
+  /* 56 */ 60, 61, 255, 255, 255, 255, 255, 255,
+
+  /* 64 */ 255, /* A - Z */ 0, 1, 2, 3, 4, 5, 6,
+  /* 72 */ 7, 8, 9, 10, 11, 12, 13, 14,
+  /* 80 */ 15, 16, 17, 18, 19, 20, 21, 22,
+  /* 88 */ 23, 24, 25, 255, 255, 255, 255, 255,
+  /* 96 */ 255, /* a - z */ 26, 27, 28, 29, 30, 31, 32,
+  /* 104 */ 33, 34, 35, 36, 37, 38, 39, 40,
+  /* 112 */ 41, 42, 43, 44, 45, 46, 47, 48,
+  /* 120 */ 49, 50, 51, 255, 255, 255, 255, 255,
+};
+// clang-format on
+
+bool base64CharacterToValue(char character, uint8_t *value) {
```

Comment:
You could make this a reference instead, and skip the `*` when assigning to it on line 3726, as well as the `&` in the uses later on.
```suggestion
bool base64CharacterToValue(char character, uint8_t &value) {
```

Reply:
nice, yes! I applied this suggestion as part of [`ae81dca` (#227)](https://github.com/fastly/js-compute-runtime/pull/227/commits/ae81dca453df75f519a3af68046348497954ab0e)